src/__tests__/exchange/swap/hooks.test.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+import { renderHook } from "@testing-library/react-hooks";
+import { useSwapProviders, initialState } from "../../../exchange/swap/hooks";
+import * as mocks from "../../../exchange/swap/mock";
+import { setEnv } from "../../../env";
+
+describe("exchange/swap/hooks", () => {
+  const OLD_ENV = process.env;
+
+  beforeEach(() => {
+    jest.resetModules();
+    jest.clearAllMocks();
+
+    // LLC internal variable
+    setEnv("MOCK", "true");
+
+    // All env variables including ones setted by LLM/LLD
+    process.env = { ...OLD_ENV };
+  });
+
+  afterAll(() => {
+    // Restore old environment
+    process.env = OLD_ENV;
+  });
+
+  test("fetches providers on mount", async () => {
+    const spy = jest.spyOn(mocks, "mockGetProviders");
+    renderHook(() => useSwapProviders());
+    expect(spy).toHaveBeenCalledTimes(1);
+  });
+
+  test("returns initial state on mount", async () => {
+    const { result } = renderHook(() => useSwapProviders());
+
+    expect(result.current).toEqual(initialState);
+  });
+
+  test("returns providers", async () => {
+    const spy = jest.spyOn(mocks, "mockGetProviders");
+    const { result, waitForNextUpdate } = renderHook(() => useSwapProviders());
+
+    await waitForNextUpdate();
+
+    expect(result.current.isLoading).toBe(false);
+    expect(result.current.error).toBe(null);
+    expect(result.current.providers.length).toBe(2);
+    expect(spy).toHaveBeenCalledTimes(1);
+  });
+
+  test("returns sorted providers", async () => {
+    const providers = ["bitfinex", "kraken", "bitstamp", "changelly", "wyre"];
```

Comment:
That would be awesome in the future to create some factories by using libraries like Fishery and faker. That will allow us to write tests faster and increase our confidence in them. I'm voluntary if needed.